src/Components/Aspire.Azure.AI.OpenAI/AspireAzureOpenAIChatClientExtensions.cs

```
@@ -0,0 +1,84 @@
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{AzureOpenAIClient, AzureOpenAIClientOptions}"/>.</param>
+    /// <param name="deploymentName">Optionally specifies the deployment name. If not specified, a value will be taken from the connection string.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.AI.OpenAI" section.</remarks>
+    public static void AddAzureOpenAIChatClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Func<ChatClientBuilder, ChatClientBuilder>? configurePipeline = null,
+        Action<AzureOpenAISettings>? configureSettings = null,
+        Action<IAzureClientBuilder<AzureOpenAIClient, AzureOpenAIClientOptions>>? configureClientBuilder = null,
+        string? deploymentName = null)
+    {
+        builder.AddAzureOpenAIClient(connectionName, configureSettings, configureClientBuilder);
+
+        builder.Services.AddSingleton(services =>
+        {
+            var chatClientBuilder = new ChatClientBuilder(services);
+            configurePipeline?.Invoke(chatClientBuilder);
+
+            deploymentName ??= GetRequiredDeploymentName(builder.Configuration, connectionName);
+
+            var innerClient = chatClientBuilder.Services
+                .GetRequiredService<AzureOpenAIClient>()
+                .AsChatClient(deploymentName);
+
+            return chatClientBuilder.Use(innerClient);
+        });
+    }
+
+    private static string GetRequiredDeploymentName(IConfiguration configuration, string connectionName)
+    {
+        string? deploymentName = null;
+
+        if (configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            var connectionBuilder = new DbConnectionStringBuilder { ConnectionString = connectionString };
+            deploymentName = (connectionBuilder[DeployentKey] ?? connectionBuilder[ModelKey]).ToString();
```

If `connectionBuilder[DeploymentKey]` might be null, what guarantees `connectionBuilder[ModelKey]` will be non-null? If it's null, the ToString will null ref

---

```suggestion
            deploymentName = (connectionBuilder[DeployentKey] ?? connectionBuilder[ModelKey])?.ToString();
```

---

Actually, it looks like DbConnectionStringBuilder's indexer will throw rather than return null if the keyword isn't found?

---

Really? I thought I checked that. Maybe I somehow didn't. But this will certainly come up when adding unit tests once the API design is agreed.